scripts/gulpfile.factory.js

```diff
@@ -271,18 +244,24 @@ ${fs
   .join("")}*/\n\n`
         )
       )
-      .pipe(dest(paths.compiled));
+      .pipe(dest(paths.temp));
   });
 
-  task("copy", () => {
-    return src(["*.js", `!${elementName}.js`], {
+  task("copy:src", () => {
+    return src(["*.js", `!${elementName}*.js`], {
       cwd: paths.source
-    }).pipe(dest(paths.compiled));
+    }).pipe(dest(paths.temp));
+  });
+
+  task("copy:compiled", () => {
+    return src(["*"], {
+      cwd: paths.temp
+    }).pipe(gulpif((file) => (files.length > 0 && gulpmatch(file, files)) || files.length === 0, dest(paths.compiled)));
```

Comment:
@kylebuch8 This says to put this file in the compiled assets folder if:
- the files array from the package.json has information in it AND the file in the pipeline for this task is in that array
- OR if there is no array or the array is empty

This sets up the default behavior that the file from the temp directory only moves to root if it's defined in the files array or if the files array doesn't exist.

Reply:
Those conditions sound good to me too.  Quick question, does gulpmatch (which I see uses vinyl-fs) support the same globbing syntax that's used in package.json's files array?  I did a few searches and wasn't able to figure that out.  There are a lot of globbing syntaxes and package.json's is pretty weird.

Reply:
gulpmatch uses minimatch syntax which looks extremely similar to what is supported by npm and gitignore.  I'm fuzzy as to if the ?(foo) syntax works, we'll have to play that one by ear, but it's easily updated to a more explicit declaration too.